### DISCUSSION TITLE
Use in WPF

### DISCUSSION THREAD

**Discussion by joselaks** [IfcOpenShell/IfcOpenShell#2610] [General]
Is there any experience of using ifcOpenShell inside a WPF application? 
The idea is to generate an IFC viewer inside .NET 6 applications.
Best regards,
José

**Comment by aothms** [comment]
Not me personally, but I know it has been done. One quick way is to simply embed the IfcConvert.exe and interface with it from the .net app. The other alternative is to interface with the program code (through SWIG semi-automically as we do for our python module) or manually using `pinvoke`. Also have a look at https://github.com/xBimTeam/ of course which is built around a .net tech stack.
